src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -725,10 +725,13 @@ public int Width
 
         private void InitFromStreamWithSize(Stream stream, int width, int height)
         {
-            //read the icon header
-            if (stream == null || stream.Length == 0)
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
+
+            if (stream.Length == 0)
                 throw new System.ArgumentException("The argument 'stream' must be a picture that can be used as a Icon", "stream");
```

Comment:
Are localized argument strings tracked elsewhere?

Reply:
They should be -- the mono version doesn't use them anywhere... I will file something.

Reply:
https://github.com/dotnet/corefx/issues/23749